File: fixtures/showcase.rs
#[derive(Debug)]
struct ThemePreference<'a> {
    appearance: &'a str,
    contrast: f32,
}

fn describe_theme(preference: &ThemePreference<'_>) -> String {
    format!("{}: {:.1}", preference.appearance, preference.contrast.max(4.5))
}

File: fixtures/showcase.php
<?php

namespace Everforest;

final class ThemePreference
{
    public function __construct(
        public readonly string $mode,
        public readonly float $contrast,
    ) {}

    public function readableContrast(): float
    {
        return max($this->contrast, 4.5);
    }
}
